compiler/src/dotty/tools/io/Path.scala

```diff
@@ -160,8 +158,17 @@ class Path private[io] (val jpath: JPath) {
     val p = parent
     if (p isSame this) Nil else p :: p.parents
   }
+
+  def ext: FileExtension = Path.fileExtension(name)
+
   // if name ends with an extension (e.g. "foo.jpg") returns the extension ("jpg"), otherwise ""
-  def extension: String = Path.extension(name)
+  def extension: String = ext.toLowerCase
+
+  def hasExtension(ext: FileExtension, exts: FileExtension*): Boolean = {
+    val myExt = this.ext
+    ext == myExt || exts.exists(_ == myExt)
+  }
+
   // compares against extensions in a CASE INSENSITIVE way.
   def hasExtension(ext: String, exts: String*): Boolean = {
```

Comment:
Deprecated?

Reply:
a bunch of these I kept because some source test used them

Reply:
I deprecated it